test/software/amazon/ion/impl/LocalSymbolTableImportAdapterTest.java

```diff
@@ -0,0 +1,250 @@
+    public void testMaxIdNoImports()
+        throws Exception
+    {
+        LocalSymbolTable delegate = localSymbolTableBuilder()
+            .setSymbols("one", "two")
+            .build();
+
+        LocalSymbolTableImportAdapter subject = LocalSymbolTableImportAdapter.of(delegate);
+
+        assertEquals(2, subject.getMaxId());
+    }
+
+    @Test
+    public void testMaxIdWithImports()
+        throws Exception
+    {
+        SymbolTable imported = mockImport("1", "2", "3");
+
+        LocalSymbolTable delegate = localSymbolTableBuilder()
+            .setSymbols("one", "two")
+            .setImportedTables(imported)
+            .build();
+
+        LocalSymbolTableImportAdapter subject = LocalSymbolTableImportAdapter.of(delegate);
+
+        assertEquals(5, subject.getMaxId());
+    }
+
+    @Test
+    public void testImportedMaxIdNoImports()
+        throws Exception
+    {
+        LocalSymbolTable delegate = localSymbolTableBuilder()
+            .setSymbols("one", "two")
+            .build();
+
+        LocalSymbolTableImportAdapter subject = LocalSymbolTableImportAdapter.of(delegate);
+
+        assertEquals(0, subject.getImportedMaxId());
+    }
+
+    @Test
+    public void testImportedMaxIdWithImports()
+        throws Exception
+    {
+        SymbolTable imported = mockImport("1", "2", "3");
+
+        LocalSymbolTable delegate = localSymbolTableBuilder()
+            .setSymbols("one", "two")
+            .setImportedTables(imported)
+            .build();
+
+        LocalSymbolTableImportAdapter subject = LocalSymbolTableImportAdapter.of(delegate);
+
+        assertEquals(3, subject.getImportedMaxId());
+    }
+
+    @Test(expected = ReadOnlyValueException.class)
+    public void testIntern()
+        throws Exception
+    {
+        LocalSymbolTableImportAdapter subject = LocalSymbolTableImportAdapter.of(localSymbolTableBuilder().build());
+
+        subject.intern("");
+    }
+
+    @Test
+    public void testFind()
+        throws Exception
+    {
+        SymbolTable imported = mockImport("onImport");
+
+        LocalSymbolTable delegate = localSymbolTableBuilder()
+            .setImportedTables(imported)
+            .setSymbols("onDelegate")
+            .build();
+
+        LocalSymbolTableImportAdapter subject = LocalSymbolTableImportAdapter.of(delegate);
+
+        SymbolToken onDelegateSymbol = subject.find("onDelegate");
+        assertEquals("onDelegate", onDelegateSymbol.getText());
+        assertEquals(2, onDelegateSymbol.getSid());
+
+        SymbolToken onImportSymbol = subject.find("onImport");
+        assertEquals("onImport", onImportSymbol.getText());
+        assertEquals(1, onImportSymbol.getSid());
+
+        SymbolToken unknown = subject.find("unknown");
+        assertNull(unknown);
+    }
+
+    private LocalSymbolTableMockBuilder localSymbolTableBuilder()
+    {
+        return new LocalSymbolTableMockBuilder(system());
+    }
+
+    private SymbolTable mockImport(String... symbols)
```

Comment:
It seems simple to change this a bit to remove the Mockito dependency too.

Reply:
This one is a bit abusive since it's using mockito to create stubs and it'd be better to use real values, will change this